src/net_processing.cpp

```diff
@@ -468,6 +459,50 @@ static CNodeState *State(NodeId pnode) EXCLUSIVE_LOCKS_REQUIRED(cs_main) {
     return &it->second;
 }
 
+/**
+ * Data structure for an individual peer. This struct is not protected by
+ * cs_main since it does not contain validation-critical data.
+ *
+ * Memory is owned by shared pointers and this object is destructed when
+ * the refcount drops to zero.
+ *
+ * TODO: move most members from CNodeState to this structure.
+ * TODO: move remaining application-layer data members from CNode to this structure.
+ */
+struct Peer {
+    /** Same id as the CNode object for this peer */
+    const NodeId m_id;
```

Comment:
337528b ISTM this type is an `int64_t` defined in net.h; should we initialize a default here for safety, like for the other members `m_misbehavior_score{0}` and `m_should_discourage{false}` added in 36ac41a

Reply:
Sure. This gets set in the initializer list of the only ctor, but there's no harm in adding this.

Reply:
It is impossible to not initialize a const member, so I'd prefer to keep it as is, but not strong opinion